lib/xenops_server.ml

```diff
@@ -2319,27 +2370,19 @@ module VM = struct
     let memory_limit = List.assoc "memory_limit" cookies |> Int64.of_string in
     Debug.with_thread_associated dbg
       (fun () ->
-         let is_localhost, id = Debug.with_thread_associated dbg
+         let id, final_id = Debug.with_thread_associated dbg
```

Comment:
This `Debug.with_thread_associated dbg` actually looks redundant, since there is one two lines above.